stix_shifter_modules/cybereason/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,471 @@
+                previous_query_path = previous_query["queryPath"][-1]
+                previous_requested_type = previous_query_path["requestedType"]
+                current_query_path = current_query["queryPath"][0]
+                current_requested_type = current_query_path["requestedType"]
+                if current_requested_type == previous_requested_type:
+                    QueryStringPatternTranslator._merge_similar_element(current_query, previous_query, merged_query)
+                elif current_requested_type in self.config_map["linked_fields"][previous_requested_type].keys():
+                    self._merge_linked_element(previous_query,
+                                               current_query, merged_query)
+        if not merged_query:
+            raise LinkNotFoundException('Link is not found between elements')
+        return merged_query
+
+    def _parse_mapped_fields(self, comparator, value, mapped_fields_array, qualifier):
+        """
+        Creates queries based on combined comparison expression.
+        Created queries will be updated in combined_query
+        :param comparator: str , comparison expression operator type
+        :param value: formatted list
+        :param mapped_fields_array : list of mapped fields
+        """
+        current_query = self._create_single_comparison_query(comparator, value, mapped_fields_array, qualifier)
+        if not self.qualified_queries[-1]:
+            self.qualified_queries[-1] = current_query
+        else:
+            previous_query = self.qualified_queries.pop()
+            merged_query = self._and_operator_query(previous_query, current_query)
+            self.qualified_queries.append(merged_query)
+
+    def __eval_observation_expression(self, expression, qualifier):
+        """
+        Function for parsing observation expression
+        :param expression: expression object
+        :param qualifier: qualifier
+        """
+        self.qualified_queries.append([])
+        self._parse_expression(expression.comparison_expression, qualifier)
+        if len(self.qualified_queries) > 1:
+            current_query = self.qualified_queries.pop()
+            previous_query = self.qualified_queries.pop()
+            merged_query = self._and_operator_query(previous_query, current_query)
+            self.qualified_queries.append(merged_query)
+
+    def _parse_expression(self, expression, qualifier=None):
+        """
+        Formation of Cybereason query from ANTLR parsing expression
+        :param expression: expression object, ANTLR parsed expression object
+        :param qualifier: str, default in None
+        :return :None or str
+        """
+        if isinstance(expression, ComparisonExpression):  # Base Case
+            stix_object, stix_field = expression.object_path.split(':')
+            mapped_fields_array = self.dmm.map_field(stix_object, stix_field)
+            comparator = self.comparator_lookup[expression.comparator]
```

Comment:
change this to `comparator = self.comparator_lookup[str(expression.comparator)]`

Reply:
Updated the code as suggested.